### PR TITLE
[Prototype] Invalidate lock file when search paths in global.json is changed

### PR DESCRIPTION
**Pull request by troydai**
Addressing #2671 

/cc @davidfowl @yishaigalatzer @emgarten

### CODE REVIEW

**Comment by troydai** [comment]
I have a follow up change to move the search paths to the end of the lock file.

File: src/Microsoft.Dnx.Runtime/DependencyManagement/LockFile.cs [hunk 1]
```diff
@@ -31,6 +31,35 @@ public bool IsValidForProject(Project project, out string message)
                 return false;
             }
 
+            // Verify the global search paths
+            message = $"The project search paths defined in {GlobalSettings.GlobalFileName} have changed";
+            GlobalSettings globalSettings;
+            if (GlobalSettings.TryGetGlobalSettings(ProjectRootResolver.ResolveRootDirectory(project.ProjectDirectory), out globalSettings))
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/DependencyManagement/LockFile.cs]: We already resolve the root directory. Why not pass it in? I'd also measure the performance difference this additional file read adds. We already parse it to figure out the nuget path.

File: src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs [hunk 2]
```diff
@@ -811,10 +811,17 @@ private Task<LockFile> ReadLockFile(string projectLockFilePath)
             var lockFile = new LockFile();
             lockFile.Islocked = Lock;
 
+            // Set the search paths defined in global settings
+            GlobalSettings globalSettings;
+            if (GlobalSettings.TryGetGlobalSettings(ProjectRootResolver.ResolveRootDirectory(project.ProjectDirectory), out globalSettings))
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs]: Doesn't this get resolved elsewhere?

**Comment by troydai** [comment]
We have a new solution to the issue.

**Pull request closed without merging**